components/test-suite/build.xml

```diff
@@ -267,10 +267,12 @@ Type "ant -p" for a list of targets.
       <sysproperty key="testng.multiplier" value="${testng.multiplier}"/>
       <sysproperty key="lurawave.license" value="${lurawave.license}"/>
       <sysproperty key="testng.in-memory" value="${testng.in-memory}"/>
+      <sysproperty key="testng.threadCount" value="${testng.threadCount}"/>
       <jvmarg value="-mx${testng.memory}"/>
       <jvmarg value="-Duser.language=${user.language}"/>
       <jvmarg value="-Duser.country=${user.country}"/>
       <jvmarg value="-Dlogback.configurationFile=${logback.configurationFile}"/>
+      <jvmarg value="-javaagent:${root.dir}/jar/file-leak-detector-1.13.jar=strong"/>
```

Comment:
From http://maven.apache.org/ant-tasks/examples/dependencies.html, 

```diff
--- a/components/test-suite/build.xml
+++ b/components/test-suite/build.xml
@@ -272,7 +272,7 @@ Type "ant -p" for a list of targets.
       <jvmarg value="-Duser.language=${user.language}"/>
       <jvmarg value="-Duser.country=${user.country}"/>
       <jvmarg value="-Dlogback.configurationFile=${logback.configurationFile}"/>
-      <jvmarg value="-javaagent:${root.dir}/jar/file-leak-detector-1.13.jar=strong"/>
+      <jvmarg value="-javaagent:${org.kohsuke:file-leak-detector:jar}=strong"/>
     </testng>
     <fail if="failedTest"/>
   </target>
```

should allow to consume the JAR without committing it to the source code. Testing it locally I get an error when lauching the JVM:

```
test-automated:
   [testng] Error occurred during initialization of VM
   [testng] agent library failed to init: instrument
   [testng] objc[95933]: Class JavaLaunchHelper is implemented in both /Library/Java/JavaVirtualMachines/jdk1.8.0_111.jdk/Contents/Home/jre/bin/java (0x10378c4c0) and /Library/Java/JavaVirtualMachines/jdk1.8.0_111.jdk/Contents/Home/jre/lib/libinstrument.dylib (0x1057ba4e0). One of the two will be used. Which one is undefined.
   [testng] Failed to find Premain-Class manifest attribute in /Users/sbesson/.m2/repository/org/kohsuke/file-leak-detector/1.13/file-leak-detector-1.13.jar
   [testng] The tests failed.
```

Indeed, the manifest of the JAR is different between the one retrieved via Maven and the one checked out locally

```shell
sbesson@ls30630:bioformats ((bdccebd6da...)) $ unzip -qc jar/file-leak-detector-1.13.jar META-INF/MANIFEST.MF
Manifest-Version: 1.0
Premain-Class: org.kohsuke.file_leak_detector.AgentMain
Archiver-Version: Plexus Archiver
Built-By: kohsuke
Agent-Class: org.kohsuke.file_leak_detector.AgentMain
Can-Redefine-Classes: true
Can-Retransform-Classes: true
Created-By: Apache Maven 3.5.0
Build-Jdk: 1.8.0_172
Main-Class: org.kohsuke.file_leak_detector.Main

sbesson@ls30630:bioformats ((bdccebd6da...)) $ unzip -qc /Users/sbesson/.m2/repository/org/kohsuke/file-leak-detector/1.13/file-leak-detector-1.13.jar META-INF/MANIFEST.MF
Manifest-Version: 1.0
Archiver-Version: Plexus Archiver
Built-By: kohsuke
Created-By: Apache Maven 3.5.0
Build-Jdk: 1.8.0_172
```

Reply:
Thanks for the pointers.  The ```jar-with-dependencies``` is what was checked in, so that explains the difference.

4560e1b updates the POM to depend upon the ```jar-with-dependencies``` specifically, switches ```build.xml``` to use the relevant property, and removes the committed ```file-leak-detector-1.13.jar```.  That all seems to work locally, but we'll see what happens with the builds tonight.